### PR TITLE
feat(ui-react-native): Add theming for IconButton primitive

### PR DESCRIPTION
**Pull request by joebuono**
#### Description of changes

- Removed unused `text` property from styles
- Updated unit tests to use `@testing-library/react-native`

#### Checklist

- [X] PR description included
- [X] `yarn test` passes
- [X] Tests are updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-native/src/primitives/IconButton/styles.ts [hunk 1]
```diff
@@ -1,7 +1,12 @@
...
 
-export const styles: IconButtonStyles = StyleSheet.create({
-  pressed: { opacity: 0.6 },
-  text: { alignSelf: 'center' },
-});
+export const getThemedStyles = (theme: StrictTheme): IconButtonStyles => {
+  const { components, opacities } = theme.tokens;
+
+  return StyleSheet.create({
+    pressed: { opacity: opacities[60], ...components?.iconbutton.pressed },
```
Comment (calebpollman) [inline packages/react-native/src/primitives/IconButton/styles.ts:10]: Thinking we should probably also provide a `container` prop so users can apply the theme to the `Pressable` when its not pressed

Comment (ioanabrooks) [reply packages/react-native/src/primitives/IconButton/styles.ts:10]: +1 Also disabled and size.

**Comment by calebpollman** [comment]
LGTM. Just to confirm, did we check if it impacts usage in the InApp Messaging components?

**Review by ioanabrooks — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Comment by joebuono** [comment]
> LGTM. Just to confirm, did we check if it impacts usage in the InApp Messaging components?

Doesn't appear to impact InApp Messaging

**Review by ioanabrooks — APPROVED** [review]

**Pull request merged**